vault/identity_store_oidc_provider.go

```diff
@@ -277,6 +313,57 @@ func oidcProviderPaths(i *IdentityStore) []*framework.Path {
 			HelpSynopsis:    "Query OIDC configurations",
 			HelpDescription: "Query this path to retrieve the configured OIDC Issuer and Keys endpoints, response types, subject types, and signing algorithms used by the OIDC backend.",
 		},
+		{
+			Pattern: "oidc/provider/" + framework.GenericNameRegex("name") + "/authorize",
+			Fields: map[string]*framework.FieldSchema{
+				"name": {
+					Type:        framework.TypeString,
+					Description: "Name of the provider",
+				},
+				"client_id": {
+					Type:        framework.TypeString,
+					Description: "The ID of the requesting client.",
+				},
+				"scope": {
+					Type:        framework.TypeString,
+					Description: "A space-delimited, case-sensitive list of scopes to be requested. The 'openid' scope is required.",
+				},
+				"redirect_uri": {
+					Type:        framework.TypeString,
+					Description: "The redirection URI to which the response will be sent.",
+				},
+				"response_type": {
+					Type:        framework.TypeString,
+					Description: "The OIDC authentication flow to be used. The following response types are supported: 'code'",
+				},
+				"state": {
+					Type:        framework.TypeString,
+					Description: "The value used to maintain state between the authentication request and client.",
+				},
+				"nonce": {
+					Type:        framework.TypeString,
+					Description: "The value that will be returned in the ID token nonce claim after a token exchange.",
+				},
+				"max_age": {
+					Type:        framework.TypeInt,
+					Description: "The allowable elapsed time in seconds since the last time the end-user was actively authenticated.",
+				},
+			},
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ReadOperation: &framework.PathOperation{
+					Callback:                    i.pathOIDCAuthorize,
+					ForwardPerformanceStandby:   true,
+					ForwardPerformanceSecondary: false,
```

Comment:
👍 Good question

Reply:
The authorization request doesn't write anything to storage, so I don't see a need to forward the request from  secondary to primary cluster. This means that tokens issued by a provider in a given cluster should be validated within that same cluster. I believe this is a good thing, since we don't want to have to forward each request in the auth flow to the primary cluster.

We will forward the authorization request from standbys to active _within_ a cluster. We need to be able to cache and subsequently look up `authCodeCacheEntry`s for the token endpoint. We don't have a distributed cache, so we'll forward within a cluster to the active.

Reply:
Thanks, that makes sense!